docroot/modules/humsci/hs_bugherd/src/Plugin/rest/resource/BugherdResource.php

```diff
@@ -0,0 +1,447 @@
+    if (empty($issue = $this->getJiraIssue($task['external_id']))) {
+      $new_issue_created = TRUE;
+      $issue = $this->createJiraIssue($task);
+
+      $this->logger->info('New JIRA issue from bugherd. Jira: @jira, Bugherd: @bugherd', [
+        '@jira' => $issue->getKey(),
+        '@bugherd' => $task['local_task_id'],
+      ]);
+    }
+
+    // Issue needs to be updated.
+    if (!$new_issue_created) {
+      $issue->fields->setDescription($this->buildDescription($task));
+      $issue->save();
+
+      if ($task['status'] == HsBugherd::BUGHERDAPI_CLOSED) {
+        $issue->addComment($this->t('Issue closed in Bugherd by @name', ['@name' => $task['updater']['display_name']]));
+      }
+    }
+    return $issue->getKey();
+  }
+
+  /**
+   * Add a comment to JIRA from bugherd.
+   *
+   * @param array $comment
+   *   Bugherd Comment data.
+   * @param array $task
+   *   Bugherd Task data.
+   *
+   * @return \Drupal\Core\StringTranslation\TranslatableMarkup|string
+   *   Updated issue or message.
+   *
+   * @throws \Exception
+   */
+  protected function bugherdTaskCommented(array $comment, array $task) {
+    // Block comments from anonymous users so we don't have looping APIs.
+    if (empty($comment['user']['email'])) {
+      $this->logger->info('Anonymous comment rejected for ticket @id', ['@id' => $task['local_task_id']]);
+      return $this->t('Comment rejected from Anonymous');
+    }
+
+    if ($issue = $this->getJiraIssue($task['external_id'])) {
+      // Add the comment now.
+      if ($issue->addComment($comment['text'])) {
+        $this->logger->info('New comment sent to JIRA issue from @name. Jira: @jira, Bugherd: @bugherd', [
+          '@name' => $comment['user']['display_name'],
+          '@jira' => $issue->getKey(),
+          '@bugherd' => $task['local_task_id'],
+        ]);
+      }
+      else {
+        throw new \Exception(t('JIRA comment could not be created for task # @bugherd', ['@bugherd' => $task['local_task_id']]));
+      }
+
+      return $issue->getKey();
+    }
+
+    throw new \Exception(t('Unable to find JIRA ticket for task # @bugherd', ['@bugherd' => $task['local_task_id']]));
+  }
+
+  /**
+   * Load a specific JIRA issue.
+   *
+   * @param string $issue_id
+   *   Issue ID, ie: DEV-123.
+   *
+   * @return \biologis\JIRA_PHP_API\Issue
+   */
+  protected function getJiraIssue($issue_id) {
+    return $issue_id ? $this->jiraIssueService->load($issue_id) : FALSE;
+  }
+
+  /**
+   * Get the configured task name to be created in JIRA.
+   *
+   * @param array $task
+   *   Array of task values from bugherd.
+   *
+   * @return string
+   *   Created name.
+   */
+  protected function getTaskName(array $task) {
+    // Trim down the descrption to only 5 words so we dont clutter up JIRA with
+    // a long paragraph as the summary. Also Jira doesnt like new lines in the
+    // Summary
+    $description_words = explode(' ', trim(preg_replace("/\r|\n/", "", $task['description'])));
```

Comment:
Perhaps new lines should be replaced with a space instead of nothing? Would this not join the last and first characters of the two lines?

Reply:
yeah good point. probably should have  space